specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2025-10-01-preview/cognitiveservices.json

```diff
@@ -11258,55 +11572,340 @@
+      "type": "string",
+      "enum": [
+        "Managed",
+        "Hosted",
+        "Custom",
+        "None"
+      ],
+      "x-ms-enum": {
+        "name": "AgentDeploymentType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Managed",
+            "description": "The underlying infra is managed by the platform in the deployer's subscription"
+          },
+          {
+            "value": "Hosted",
+            "description": "The underlying infra is owned by the platform"
+          },
+          {
+            "value": "Custom",
+            "description": "The underlying infra is provisioned by the deployer (BYO)"
+          },
+          {
+            "value": "None",
+            "description": "Invalid type"
+          }
+        ]
+      }
+    },
+    "AgentDeploymentProvisioningState": {
+      "description": "Provisioning state of an agentic deployment, as an Azure resource.",
+      "type": "string",
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Creating",
+        "Updating",
+        "Deleting",
+        "None"
+      ],
+      "x-ms-enum": {
+        "name": "AgentDeploymentProvisioningState",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Succeeded",
+            "description": "The deployment was successfully completed."
+          },
+          {
+            "value": "Failed",
+            "description": "The deployment failed."
+          },
+          {
+            "value": "Canceled",
+            "description": "The deployment was canceled."
+          },
+          {
+            "value": "Creating",
+            "description": "The deployment is being created."
+          },
+          {
+            "value": "Updating",
+            "description": "The deployment is being updated."
+          },
+          {
+            "value": "Deleting",
+            "description": "The deployment is being deleted."
+          },
+          {
+            "value": "None",
+            "description": "Invalid state"
+          }
+        ]
+      }
+    },
+    "TrafficRoutingProtocol": {
+      "description": "Traffic routing protocol, used to distribute an application's inbound traffic to its deployments.",
+      "type": "string",
+      "enum": [
+        "FixedRatio",
+        "None"
+      ],
+      "x-ms-enum": {
+        "name": "TrafficRoutingProtocol",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "FixedRatio",
+            "description": "Percentage based routing"
+          },
+          {
+            "value": "None",
+            "description": "Invalid type"
+          }
+        ]
+      }
+    },
+    "TrafficRoutingRule": {
+      "description": "Represents a rule for routing traffic to a specific deployment.",
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

[ARMBlockingComment] Don't reuse top-level property names within the property bag. This could be ruleId (or something else you prefer). Same comment applies here: try to avoid specifying a generic "id". Instead choose a name and type that indicates clearly what kind of id it is.